src/WidgetBase.ts

```diff
@@ -134,23 +133,21 @@ export class WidgetBase<P extends WidgetProperties> extends Evented implements W
 		this.own(this.on('properties:changed', (evt: PropertiesChangeEvent<WidgetBase<WidgetProperties>, WidgetProperties>) => {
 			this.invalidate();
 		}));
-
-		this.setProperties(properties);
 	}
 
 	public get properties(): Readonly<P> {
 		return this._properties;
 	}
 
-	public setProperties(properties: P & { [index: string]: any }): void {
+	public setProperties(properties: P): void {
 		const diffPropertyResults: { [index: string]: PropertyChangeRecord } = {};
 		const diffPropertyChangedKeys: string[] = [];
 
 		this.bindFunctionProperties(properties);
 
 		this.diffPropertyFunctionMap.forEach((property: string, diffFunctionName: string) => {
 			const previousProperty = this.previousProperties[property];
-			const newProperty = properties[property];
+			const newProperty = (<any> properties)[property];
```

Comment:
Does anyone know a better way to do this? I don't want an index type on the object (WidgetProperties) as would like it to correctly report unknown properties...?